### DISCUSSION TITLE
Typst as a React component or Web component for preview AND download as PDF

### DISCUSSION THREAD

**Discussion by nileshtrivedi** [typst/typst#3673] [Ideas]
Is it possible to use Typst to show preview of a report in a webapp and also allow the user to download the report as PDF? It would be similar to Typst webapp but I don't need the editing capabilities because these reports would be programmatically generated by my app.

Ideally, this would be a component that runs completely inside the browser so that I don't need to deploy a backend / docker container. One complication I foresee is passing SVG files for figures and charts to this component.

jsPDF has this capability which is completely client-side: http://raw.githack.com/MrRio/jsPDF/master/index.html
I'm wondering if something similar is possible that lets me use Typst syntax.

**Comment by istudyatuni** [comment]
There are [typst.ts](https://github.com/Myriad-Dreamin/typst.ts) project, it has React integration

**Reply by fakhrirafiki** [reply]
I also waiting for the preview project
